mentoring/components/questionnaire.py

```diff
@@ -56,34 +51,36 @@ class QuestionnaireAbstractBlock(LightChild, StepMixin):
                    default=1, scope=Scope.content, enforce_type=True)
 
     valid_types = ('choices')
+    has_children = True
 
     @classmethod
-    def init_block_from_node(cls, block, node, attr):
-        block.light_children = []
-        for child_id, xml_child in enumerate(node):
+    def parse_xml(cls, node, runtime, keys, id_generator):
+        block = runtime.construct_xblock_from_class(cls, keys)
+
+        # Load XBlock properties from the XML attributes:
+        for name, value in node.items():
+            setattr(block, name, value)
+
+        for xml_child in node:
             if xml_child.tag == 'question':
                 block.question = xml_child.text
             elif xml_child.tag == 'message' and xml_child.get('type') == 'on-submit':
                 block.message = (xml_child.text or '').strip()
-            else:
-                cls.add_node_as_child(block, xml_child, child_id)
-
-        for name, value in attr:
-            setattr(block, name, value)
+            elif xml_child.tag is not etree.Comment:
```

Comment:
@bradenmacdonald this is not exactly a correct way to fix this. In previous version it was fixed like this: https://github.com/open-craft/xblock-mentoring/commit/1a6e2092a361f0befff18d07607517d437046e3b#diff-23a9d1f1d9cda06ac8fa67c8c2f4589dR130-131

Now the code that deserializes XML into mentoring XBlocks are in main XBlock repository: https://github.com/edx/XBlock/blob/9c634481dfc85a17dcb3351ca232d7098a38e10e/xblock/runtime.py#L685

This is probably a good idea to teach upstream XBlock to play nicely with XML comments